Build failure: vllm

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Works for CUDA but not for ROCM

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
[...]
copying vllm/plugins/lora_resolvers/README.md -> build/lib.linux-x86_64-cpython-313/vllm/plugins/lora_resolvers
running build_ext
Using MAX_JOBS=24 as the number of jobs.
-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/ag7fc58qzff4j1sgd7v98irhcr6vcg8v-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Build type: RelWithDebInfo
-- Target device: rocm
-- Found Python: /nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/bin/python3.13 (found version "3.13.9") found components: Interpreter Development.Module Development.SABIModule
-- Found python matching: /nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/bin/python3.13.
-- ROCM_PATH environment variable is not set and /opt/rocm does not exist.
Building without ROCm support.
CMake Warning at /nix/store/lmcbv6q3n5kdicjgp6wxp77x9dxbsxcb-python3.13-torch-2.9.0-dev/share/cmake/Torch/TorchConfig.cmake:22 (message):
  static library kineto_LIBRARY-NOTFOUND not found.
Call Stack (most recent call first):
  /nix/store/lmcbv6q3n5kdicjgp6wxp77x9dxbsxcb-python3.13-torch-2.9.0-dev/share/cmake/Torch/TorchConfig.cmake:125 (append_torchlib_if_found)
  CMakeLists.txt:84 (find_package)


-- Found Torch: /nix/store/nz4sh0wnais5dalj4lfby5r5ply7p9ma-python3.13-torch-2.9.0-lib/lib/libtorch.so
CMake Error at CMakeLists.txt:134 (message):
  Can't find CUDA or HIP installation.


-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/nix/store/sripdnnbs3vq9fk8hbrx63x1ll016lpg-python3.13-pyproject-hooks-1.2.0/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
    main()
    ~~~~^^
  File "/nix/store/sripdnnbs3vq9fk8hbrx63x1ll016lpg-python3.13-pyproject-hooks-1.2.0/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 373, in main
    json_out["return_val"] = hook(**hook_input["kwargs"])
                             ~~~~^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/sripdnnbs3vq9fk8hbrx63x1ll016lpg-python3.13-pyproject-hooks-1.2.0/lib/python3.13/site-packages/pyproject_hooks/_in_process/_in_process.py", line 280, in build_wheel
    return _build_backend().build_wheel(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        wheel_directory, config_settings, metadata_directory
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/build_meta.py", line 432, in build_wheel
    return _build(['bdist_wheel'])
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/build_meta.py", line 423, in _build
    return self._build_with_temp_dir(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        cmd,
        ^^^^
    ...<3 lines>...
        self._arbitrary_args(config_settings),
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/build_meta.py", line 404, in _build_with_temp_dir
    self.run_setup()
    ~~~~~~~~~~~~~~^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/build_meta.py", line 317, in run_setup
    exec(code, locals())
    ~~~~^^^^^^^^^^^^^^^^
  File "<string>", line 667, in <module>
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/__init__.py", line 115, in setup
    return distutils.core.setup(**attrs)
           ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/core.py", line 186, in setup
    return run_commands(dist)
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
    dist.run_commands()
    ~~~~~~~~~~~~~~~~~^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py", line 1002, in run_commands
    self.run_command(cmd)
    ~~~~~~~~~~~~~~~~^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py", line 1102, in run_command
    super().run_command(command)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
    ~~~~~~~~~~~^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/command/bdist_wheel.py", line 370, in run
    self.run_command("build")
    ~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
    self.distribution.run_command(command)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py", line 1102, in run_command
    super().run_command(command)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
    ~~~~~~~~~~~^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/cmd.py", line 357, in run_command
    self.distribution.run_command(command)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py", line 1102, in run_command
    super().run_command(command)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/dist.py", line 1021, in run_command
    cmd_obj.run()
    ~~~~~~~~~~~^^
  File "<string>", line 283, in run
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/command/build_ext.py", line 96, in run
    _build_ext.run(self)
    ~~~~~~~~~~~~~~^^^^^^
  File "/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_distutils/command/build_ext.py", line 368, in run
    self.build_extensions()
    ~~~~~~~~~~~~~~~~~~~~~^^
  File "<string>", line 245, in build_extensions
  File "<string>", line 223, in configure
  File "/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '/build/source', '-G', 'Ninja', '-DCMAKE_BUILD_TYPE=RelWithDebInfo', '-DVLLM_TARGET_DEVICE=rocm', '-DVLLM_PYTHON_EXECUTABLE=/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/bin/python3.13', '-DVLLM_PYTHON_PATH=/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/lib/python3.13/site-packages:/nix/store/hcysdx4qqvd41ydilcmr79g82ayhrzml-python3.13-wheel-0.46.1/lib/python3.13/site-packages:/nix/store/ff4rnarxwsmvpxwzs2jlgmhw1wgl0brx-python3.13-packaging-25.0/lib/python3.13/site-packages:/nix/store/kl5h6ywh2k0by5cwmjc9a6pm7l4aqs2j-python3.13-installer-0.7.0/lib/python3.13/site-packages:/nix/store/rs9wy0vknl0vbmqvm6ai80ibx0q83dnj-python3.13-cmake-4.1.2/lib/python3.13/site-packages:/nix/store/n5wfyqn7vwps9708k4nsdplcmihqng2p-python3.13-jinja2-3.1.6/lib/python3.13/site-packages:/nix/store/6kczbzdd1070qzgwcimjxhw94rrnyw1a-python3.13-markupsafe-3.0.3/lib/python3.13/site-packages:/nix/store/q2ys67zpirl13q89pyy4wa94v5xb7i9k-python3.13-ninja-1.13.1/lib/python3.13/site-packages:/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages:/nix/store/78scyafcrqs7cp27k6m6mx61bsaa0796-python3.13-setuptools-scm-9.0.1/lib/python3.13/site-packages:/nix/store/lgpvg0fx9y9gh9mfn32yz3qk1cgajym1-python3.13-typing-extensions-4.15.0/lib/python3.13/site-packages:/nix/store/1n95l7wzfyv7ybhkscbn44xvd1djzgdp-python3.13-astunparse-1.6.3/lib/python3.13/site-packages:/nix/store/blq1wr4vmx16nvy6gzyibqmsrnl2vx7m-python3.13-six-1.17.0/lib/python3.13/site-packages:/nix/store/1nb0a77b8ma9a09wmym7q91qy5qbnbxl-python3.13-expecttest-0.2.1/lib/python3.13/site-packages:/nix/store/82b38idspcs10fa0wvxr2lvw4i8y73nz-python3.13-filelock-3.20.0/lib/python3.13/site-packages:/nix/store/nhg9par67hvdh8iy23xp2699l24vzz43-python3.13-fsspec-2025.3.2/lib/python3.13/site-packages:/nix/store/hdanylzxqzx4iija73ami3g19s95gjpb-python3.13-hypothesis-6.136.9/lib/python3.13/site-packages:/nix/store/mg7rqd30rciwlnzr8pzl67nk3gdrckkm-python3.13-attrs-25.3.0/lib/python3.13/site-packages:/nix/store/rvl5xym4328ly59pw9zv7fq6ypamyirl-python3.13-sortedcontainers-2.4.0/lib/python3.13/site-packages:/nix/store/zz34v85kggbxhhz12202s05z7dvlvijp-python3.13-networkx-3.5/lib/python3.13/site-packages:/nix/store/0xagc3jmn7xrc1vismlwwn8rq5ckd8zl-python3.13-psutil-7.1.2/lib/python3.13/site-packages:/nix/store/418as99iprxpjn501q09bixp7k3ab34p-python3.13-pyyaml-6.0.3/lib/python3.13/site-packages:/nix/store/yg71z65c1vi2wj463gapkwd1ckgxh7h3-python3.13-requests-2.32.5/lib/python3.13/site-packages:/nix/store/pxzpsxysk4247z6pxar3jng8z150vf5a-python3.13-certifi-2025.07.14/lib/python3.13/site-packages:/nix/store/5jhiwz03zyib40726a55zqq1padhgn3w-python3.13-charset-normalizer-3.4.3/lib/python3.13/site-packages:/nix/store/sd9ny7hxa003mxpxvcqhrhdq55r2rpwh-python3.13-idna-3.11/lib/python3.13/site-packages:/nix/store/r8djhy65qwv66licydaf0lfjxvmxqj9q-python3.13-urllib3-2.5.0/lib/python3.13/site-packages:/nix/store/4fchkl9afqakrw2b5xm2q107c4rhic8j-python3.13-sympy-1.14.0/lib/python3.13/site-packages:/nix/store/8b8d8wapsyrkbx7crl9949g8vpvy9r10-python3.13-mpmath-1.3.0/lib/python3.13/site-packages:/nix/store/lnz395nrvj4xmx1bsd25ca6hg155q2yj-python3.13-types-dataclasses-0.6.6/lib/python3.13/site-packages:/nix/store/96vld9s60486ifj2pq1v3gdbq19gx145-python3.13-pillow-12.0.0/lib/python3.13/site-packages:/nix/store/vd3x2zfn6nlzlbd74ki07qfappljkayi-python3.13-tensorboard-2.20.0/lib/python3.13/site-packages:/nix/store/aljrinr2zwiimwjgkl5xr2y6syih8l58-python3.13-absl-py-2.3.1/lib/python3.13/site-packages:/nix/store/s39jk0ciw7ibkfz0ib33jgwmcsbf9kkm-python3.13-protobuf-6.33.0/lib/python3.13/site-packages:/nix/store/6ani3fmirywvkag62pjm0ilv29b3rnsb-python3.13-grpcio-1.76.0/lib/python3.13/site-packages:/nix/store/vdrrrfvlqhkbirw8087j8m9177ph9jwz-python3.13-markdown-3.9.0/lib/python3.13/site-packages:/nix/store/skjbqivjxmns0p04s3bvfpbv3xmfaj9k-python3.13-numpy-2.3.4/lib/python3.13/site-packages:/nix/store/hivfx01z9j1s81i9aw5vpgl6ynwr61z7-python3.13-tensorboard-data-server-0.7.2/lib/python3.13/site-packages:/nix/store/i81zscnj0b674zpb6zb3b407lsw785ir-python3.13-werkzeug-3.1.3/lib/python3.13/site-packages:/nix/store/nn2qv7n47qfhqib2fmd6g5k46vmbsajl-python3.13-standard-imghdr-3.13.0/lib/python3.13/site-packages:/nix/store/zil3q3msap3l0bpvjmrnsw1ia4xslsnz-python3.13-pybind11-3.0.1/lib/python3.13/site-packages:/nix/store/2sxmh3qc6c7pvzx87c1brk699h6blbag-python3.13-triton-3.5.0/lib/python3.13/site-packages:/nix/store/khj8i5pjvbn2kd8ss94rpfjhda650gyi-python3.13-torch-2.9.0/lib/python3.13/site-packages:/nix/store/l2xngvsdimm3kcmq9ibsblvhrp5qfwl9-python3.13-cffi-2.0.0/lib/python3.13/site-packages:/nix/store/nl10ppfaqvwlbxp0p9wjgfcl9kyzrkbm-python3.13-pycparser-2.23/lib/python3.13/site-packages:/nix/store/x0y61nbj2srxl5nrc842gm0vc8x6yzjd-python3.13-aioprometheus-unstable-2023-03-14/lib/python3.13/site-packages:/nix/store/54c3jicwviln5kxh2w36gaqmxpkma4k8-python3.13-orjson-3.11.3/lib/python3.13/site-packages:/nix/store/w4s9g2jdwnfxgd3m2pvj4qfrj9yzyvaf-python3.13-quantile-python-1.1/lib/python3.13/site-packages:/nix/store/hi4d538az5hpf6i8029xfbhf38gm87x7-python3.13-blake3-1.0.8/lib/python3.13/site-packages:/nix/store/9irlwycxfw7acs60m0fzdcfs05qx95s0-python3.13-cachetools-6.1.0/lib/python3.13/site-packages:/nix/store/a246d2wqx1bi6h5csffgixkvg29sdp0b-python3.13-cbor2-5.7.0/lib/python3.13/site-packages:/nix/store/9z3mqyvqng6mlggss2z4y1myl6j3isgp-python3.13-depyf-0.20.0/lib/python3.13/site-packages:/nix/store/cnq6xy93456kj4hqnrdm5138f3dks91m-python3.13-astor-0.8.1/lib/python3.13/site-packages:/nix/store/nyg7j873sc4jpg0qqf88imn6p8gd9451-python3.13-dill-0.4.0/lib/python3.13/site-packages:/nix/store/dhzw8dhavcds4i5a9vs4q6f9vr2g71ry-python3.13-fastapi-0.116.1/lib/python3.13/site-packages:/nix/store/69r7rbscppqfw5j0wfq4xdrw7s7z1h0b-python3.13-starlette-0.47.2/lib/python3.13/site-packages:/nix/store/mhq2xnxhkqz4q6b0hxvjpxm8b2cz4qyk-python3.13-anyio-4.11.0/lib/python3.13/site-packages:/nix/store/sk96kc7da77brmlazk4rck4p0fhmpzg2-python3.13-sniffio-1.3.1/lib/python3.13/site-packages:/nix/store/7m1n2rqb316jd4qdyn51m0jcgdpmrbyh-python3.13-pydantic-2.11.7/lib/python3.13/site-packages:/nix/store/zdypmz6v71x17w5bkrchqyj6bnxlhgfb-python3.13-annotated-types-0.7.0/lib/python3.13/site-packages:/nix/store/1hzg98lcbzwq8193ikp2s0q0cnn2b3k8-python3.13-pydantic-core-2.33.2/lib/python3.13/site-packages:/nix/store/3n1x675757b1vihj2886plgn0lc68v58-python3.13-typing-inspection-0.4.2/lib/python3.13/site-packages:/nix/store/nr85ghld3qy758273vdym5wafkdfxw27-python3.13-llguidance-1.3.0/lib/python3.13/site-packages:/nix/store/51chy8s28h1pxshv9qw7ii88mqq9zhjf-python3.13-lm-format-enforcer-0.11.3/lib/python3.13/site-packages:/nix/store/bgaf80whz60rz2xh4rsjww7ipbhnvsfb-python3.13-interegular-0.3.3/lib/python3.13/site-packages:/nix/store/06jib7cnbisf90zp63rb054rkxf4q4y0-python3.13-openai-2.7.2/lib/python3.13/site-packages:/nix/store/6fkvhj0c97a3482d3qpk3by6khiv8677-python3.13-distro-1.9.0/lib/python3.13/site-packages:/nix/store/x0igjmp3qpa2kxnsbaw6k28vfdm5fngm-python3.13-httpx-0.28.1/lib/python3.13/site-packages:/nix/store/7krypblnbzxkqd40pb8q4654sd379afq-python3.13-httpcore-1.0.9/lib/python3.13/site-packages:/nix/store/6gda2ckbckylz8ac6izywhyfwwjdn3zf-python3.13-h11-0.16.0/lib/python3.13/site-packages:/nix/store/jxn7lak1hfwc5f598lv22d52rwmym1vz-python3.13-jiter-0.11.1/lib/python3.13/site-packages:/nix/store/fn48895pnagck2pj086bc7nw5dhihzqn-python3.13-tqdm-4.67.1/lib/python3.13/site-packages:/nix/store/mm436l417p4fkclh5mxyx2gwdmpr7m5m-python3.13-aiohttp-3.13.2/lib/python3.13/site-packages:/nix/store/14wfnklffl6kwic3vbl18r09360d147c-python3.13-aiohappyeyeballs-2.6.1/lib/python3.13/site-packages:/nix/store/9kvys9ha87sgiyks6rl7kkiffixrds5d-python3.13-aiosignal-1.4.0/lib/python3.13/site-packages:/nix/store/mgm0jjjv6kb8g2c6vbw71pi822zjwia1-python3.13-frozenlist-1.7.0/lib/python3.13/site-packages:/nix/store/j0higmsf1krdc50l4lw3pby1pqcc0jal-python3.13-multidict-6.7.0/lib/python3.13/site-packages:/nix/store/q5kyyql30rydwhs385na0b3mzp2j34lc-python3.13-propcache-0.3.2/lib/python3.13/site-packages:/nix/store/9v72y8kjcvpfkz577z22p4ah2qaxac5a-python3.13-yarl-1.22.0/lib/python3.13/site-packages:/nix/store/rvvj706nfrpf4knh8yhzqrh7zwc1wlvi-python3.13-aiodns-3.5.0/lib/python3.13/site-packages:/nix/store/g5inrf8jzkm2zi42ip921jy0kqpy78hm-python3.13-pycares-4.9.0/lib/python3.13/site-packages:/nix/store/zrzmx5dcxlyah1ap95lmnipfvanim4l0-python3.13-backports-zstd-1.0.0/lib/python3.13/site-packages:/nix/store/f9q0qawx2b5b5wwc1v8jn7gg7xjry97s-python3.13-brotli-1.1.0/lib/python3.13/site-packages:/nix/store/n6xpmzndaijl03vf02vvxbfyddv8gfdq-python3.13-httpx-aiohttp-0.1.9/lib/python3.13/site-packages:/nix/store/fixmirg5xnw7carz93ys607crbmi6qk1-python3.13-websockets-15.0.1/lib/python3.13/site-packages:/nix/store/ycg4dln9mvz3vvc7cvd9zpwlag4hpigc-python3.13-sounddevice-0.5.2/lib/python3.13/site-packages:/nix/store/5zgrksz59hk12b2vxy0a45bnjqb90l1h-python3.13-opencv-python-headless-4.12.0/lib/python3.13/site-packages:/nix/store/mi8cx9cnykcw5gvfxbspq3xfnj02qxbm-opencv-4.12.0/lib/python3.13/site-packages:/nix/store/bi7mn9nqi67kz3q7x2fxyrwgv7hksnas-python3.13-outlines-1.2.3/lib/python3.13/site-packages:/nix/store/4jkziip4hllfn9wcdp9i714ffiycj542-python3.13-airportsdata-20251008/lib/python3.13/site-packages:/nix/store/awy476dwpg2igicqmrxabnc3i5ky1zbh-python3.13-cloudpickle-3.1.1/lib/python3.13/site-packages:/nix/store/dj9j271fn6wzr95jj5dh101i0hrjpwrp-python3.13-datasets-4.0.0/lib/python3.13/site-packages:/nix/store/3hxacrxhml65mcz0cpd8l2s6n0g52y89-python3.13-huggingface-hub-0.35.3/lib/python3.13/site-packages:/nix/store/rxrm3r8ss3mf13lm0qv2dyll68p42ddp-python3.13-hf-xet-1.1.10/lib/python3.13/site-packages:/nix/store/50vg0jwir9q39499xblwmzc3r41lh469-python3.13-multiprocess-0.70.18/lib/python3.13/site-packages:/nix/store/syi4rhx0vdlxdhw132d24qv73474xzmk-python3.13-pandas-2.3.1/lib/python3.13/site-packages:/nix/store/7djcjs47gvjri9q8y7iqc7w8822lfhs6-python3.13-python-dateutil-2.9.0.post0/lib/python3.13/site-packages:/nix/store/7kbh1r8fhhzl5l2q8ijgsd84mizglfg2-python3.13-pytz-2025.2/lib/python3.13/site-packages:/nix/store/qs4c1f15r1acbxxm3qcks2vxkvcfzpsc-python3.13-tzdata-2025.2/lib/python3.13/site-packages:/nix/store/6plx13l2j3alnplhzb064rjcnwfgs0s5-python3.13-pyarrow-20.0.0/lib/python3.13/site-packages:/nix/store/8is0jffax958b004nv93ai3xr7mshs9d-python3.13-responses-0.25.7/lib/python3.13/site-packages:/nix/store/rpjqfn93ilivc04yjisyf69jpn3c8jkp-python3.13-types-pyyaml-6.0.12.20250516/lib/python3.13/site-packages:/nix/store/sg5nbynwmwjpg4kiq5kfxzlssvh1w2r8-python3.13-types-toml-0.10.8.20240310/lib/python3.13/site-packages:/nix/store/aq5v5b5qba3khi8qniv1y7q3jk73p3v9-python3.13-xxhash-3.5.0/lib/python3.13/site-packages:/nix/store/hsxbypnfwsrr7lgvlf0viwlyd0x6nmq2-python3.13-diskcache-5.6.3/lib/python3.13/site-packages:/nix/store/i7ccdwc1d88ywhghqj9n6zdi6sryn8iv-python3.13-genson-1.3.0/lib/python3.13/site-packages:/nix/store/dcyk0prg2xqw8q4kmi4x2g4rm33j63lv-python3.13-iso3166-2.1.1/lib/python3.13/site-packages:/nix/store/nq6ggid6rqj9grbshvh4icybdb71ivl4-python3.13-jsonschema-4.25.0/lib/python3.13/site-packages:/nix/store/hyf11ywhxand1yq4iczd0hzxmb6mm07n-python3.13-jsonpath-ng-1.7.0/lib/python3.13/site-packages:/nix/store/iik6nld6j6kbqbpgnywz1n18i7cax7vb-python3.13-ply-3.11/lib/python3.13/site-packages:/nix/store/0bylz2v4gxk6fqzvg2wskrzvi3xdavgp-python3.13-jsonschema-specifications-2025.4.1/lib/python3.13/site-packages:/nix/store/mygvx4b91pq4cwiq25hrcnwm39y0q3ln-python3.13-referencing-0.36.2/lib/python3.13/site-packages:/nix/store/y7bqxil0ifa1xyg3qq5v89c23n8ji4q2-python3.13-rpds-py-0.25.0/lib/python3.13/site-packages:/nix/store/mrjhi0n40njgzdvcfmrnny6hqc69dmka-python3.13-lark-1.2.2/lib/python3.13/site-packages:/nix/store/j1fha5k1qm9xlvshraisiq9jqlgclhpg-python3.13-regex-2025.9.18/lib/python3.13/site-packages:/nix/store/27mzzq4ignjxh7xi1jn3wxh11lf1azg8-python3.13-nest-asyncio-1.6.0/lib/python3.13/site-packages:/nix/store/2npslc3q4z7a9h4x4ln77lqgxc4ysyvn-python3.13-outlines-core-0.2.11/lib/python3.13/site-packages:/nix/store/0m66w6b9jfzf8107kclpcs3jigw6hyki-python3.13-pycountry-24.6.1/lib/python3.13/site-packages:/nix/store/2hq07xrxaw1n9fpk6srz0w0cfpbqcixk-python3.13-transformers-4.57.1/lib/python3.13/site-packages:/nix/store/lvm1h8fprbkhihn7m9vc11ghmybm9jwi-python3.13-tokenizers-0.22.1/lib/python3.13/site-packages:/nix/store/caknnc2xl0bh6lkg2lyblbjlifmqps87-python3.13-safetensors-0.6.2/lib/python3.13/site-packages:/nix/store/npn7zkvbyjw0jrqm2lfh3zj9aiqfiq8k-python3.13-prometheus-fastapi-instrumentator-7.1.0/lib/python3.13/site-packages:/nix/store/1irnid3dnpf00lfjz6wnki5n6gplaxcg-python3.13-prometheus-client-0.22.1/lib/python3.13/site-packages:/nix/store/2n3f80s2dg6f1dz6sjmdwjmif0nzz1gb-python3.13-py-cpuinfo-9.0.0/lib/python3.13/site-packages:/nix/store/qdrfsmzs1zibv3zjlmcbn6b3ixq8xlcs-python3.13-pybase64-1.4.2/lib/python3.13/site-packages:/nix/store/z8x54f1xmx8i6ms0q6dfaclbj6bz0q16-python3.13-python-json-logger-3.3.0/lib/python3.13/site-packages:/nix/store/s0zyp69rb7vqjxmw3zlixqzb6kmw5k99-python3.13-python-multipart-0.0.20/lib/python3.13/site-packages:/nix/store/fdkh4hbnsmi1wkmmvpq0kr2618nzqy3f-python3.13-pyzmq-27.0.1/lib/python3.13/site-packages:/nix/store/pvdhh33w9g88frnyab1kksh3kzy8rbhz-python3.13-ray-2.51.1/lib/python3.13/site-packages:/nix/store/alzyma954kdvpwhhiil5ps3311pk6m02-python3.13-click-8.2.1/lib/python3.13/site-packages:/nix/store/p6w8g2n7frnijflp4jbm0zgbm8hwnrzb-python3.13-msgpack-1.1.1/lib/python3.13/site-packages:/nix/store/2jw7rmwslnflfpb26dgm942hb5frpbjf-python3.13-watchfiles-1.0.5/lib/python3.13/site-packages:/nix/store/wfcw2kr59vaikpnvvz01d67dgj2h86gl-python3.13-sentencepiece-0.2.1/lib/python3.13/site-packages:/nix/store/4la5k7rymwqp9ba9hgz7an85v5c3zs3l-python3.13-tiktoken-0.9.0/lib/python3.13/site-packages:/nix/store/1dsp32j3a7f0ghx8a2alb20zz1dd09lh-python3.13-blobfile-3.0.0/lib/python3.13/site-packages:/nix/store/8m8lgrqrh2jsfglp5928hmkgvkcnv5sb-python3.13-pycryptodomex-3.23.0/lib/python3.13/site-packages:/nix/store/bmhyh54y489qiw5gdhfx54ji4kcfmlzb-python3.13-lxml-6.0.2/lib/python3.13/site-packages:/nix/store/6ja9cll3kkszp6wam8pwp940wpp8gc1q-python3.13-msgspec-0.19.0/lib/python3.13/site-packages:/nix/store/d3vzd44bc3qiw3hvp1i5ihi4w0y7zask-python3.13-gguf-0.17.1/lib/python3.13/site-packages:/nix/store/xcdfijqaa5fw0fdiqil49q6kwrq3lhb3-python3.13-poetry-core-2.2.1/lib/python3.13/site-packages:/nix/store/ggjbh2h9wv0wqan6fa19mk216c69zfcz-python3.13-einops-0.8.1/lib/python3.13/site-packages:/nix/store/50irpwi8m6zjaarmv3gj2kpqvs1g6nvl-python3.13-importlib-metadata-8.7.0/lib/python3.13/site-packages:/nix/store/1wqdwljylmbcdrkpf3h57qxb5ab03crv-python3.13-toml-0.10.2/lib/python3.13/site-packages:/nix/store/fmyxn270zaw03cpsyp5gam7j9xkca1b8-python3.13-zipp-3.23.0/lib/python3.13/site-packages:/nix/store/cki014xa8hsrp96n5ajjdbqmlqzwrjnj-python3.13-partial-json-parser-0.2.1.1.post6/lib/python3.13/site-packages:/nix/store/jybkvjp5g622pbmz3gww91hm8d3gw7nz-python3.13-compressed-tensors-0.11.0/lib/python3.13/site-packages:/nix/store/kibpabrzs1n7ykxb239pvc47vzg28185-python3.13-frozendict-2.4.6/lib/python3.13/site-packages:/nix/store/3cnb9xc7fvkjrcfwy6akgv2yjpin5b4k-python3.13-mistral-common-1.8.4/lib/python3.13/site-packages:/nix/store/913n1xa0cfkdq5j6pr2wf5xdama1vag2-python3.13-pydantic-extra-types-2.10.5/lib/python3.13/site-packages:/nix/store/a7fwnarj8xv4k7yx90g1maiicdvdff1f-python3.13-torchaudio-2.9.0/lib/python3.13/site-packages:/nix/store/nd5j7vrxi33lb8hk9y1zb3c0dw0izsai-python3.13-torchvision-0.24.0/lib/python3.13/site-packages:/nix/store/60i66hjgznqpmdx63c6ybv66sv1m7751-python3.13-scipy-1.16.2/lib/python3.13/site-packages:/nix/store/v3rl3h2mnjxbspr2z7gx8wq0g0v0ds28-python3.13-uvicorn-0.35.0/lib/python3.13/site-packages:/nix/store/376jknl1f2khz295csnbq338sskfacxl-python3.13-xformers-0.0.30/lib/python3.13/site-packages:/nix/store/fklpp26s4i05dw3ygjnx21qllr14fzda-python3.13-xgrammar-0.1.24/lib/python3.13/site-packages:/nix/store/11wvg8xqw4jkiknkm86piss7kp6saaig-python3.13-numba-0.62.0/lib/python3.13/site-packages:/nix/store/l60g4hxd74ymy08wqcpkl121jsjdlzfb-python3.13-llvmlite-0.45.0/lib/python3.13/site-packages:/nix/store/a4znqiv2ra9jhs9lk6wjmwlh38c35g1g-python3.13-opentelemetry-sdk-1.34.0/lib/python3.13/site-packages:/nix/store/lnqpq2vgsxk1kjcicx4130i9j975sf3k-python3.13-opentelemetry-api-1.34.0/lib/python3.13/site-packages:/nix/store/04h5vyzcp4jpx89f3y00xizzm14m6fnd-python3.13-deprecated-1.2.18/lib/python3.13/site-packages:/nix/store/smn567nl6w2ppfmxs9ncwsx254xd7c5m-python3.13-wrapt-1.17.2/lib/python3.13/site-packages:/nix/store/1xsppjxy9b2pfs6495967i7668ijyhmx-python3.13-opentelemetry-semantic-conventions-0.55b0/lib/python3.13/site-packages:/nix/store/ngdj7hglad6lbk6hh128s3fmy8iz3xrg-python3.13-opentelemetry-exporter-otlp-1.34.0/lib/python3.13/site-packages:/nix/store/njpqmiqqgzgj1rg1mr9rm5lz9q9csci7-python3.13-opentelemetry-exporter-otlp-proto-grpc-1.34.0/lib/python3.13/site-packages:/nix/store/9b7wc8gmqasjz0bk9yirxm3qnn34xp5r-python3.13-googleapis-common-protos-3.31.3/lib/python3.13/site-packages:/nix/store/n5z7xya26gcfqi7apik7azh11k5bhhcl-python3.13-opentelemetry-exporter-otlp-proto-common-1.34.0/lib/python3.13/site-packages:/nix/store/p43149fb9wwhmzhsni3qjd77ig9ky2rs-python3.13-opentelemetry-proto-1.34.0/lib/python3.13/site-packages:/nix/store/zac452rn0lgmnaqkhl69svadz1chbh63-python3.13-opentelemetry-exporter-otlp-proto-http-1.34.0/lib/python3.13/site-packages:/nix/store/6lv5p98ygk6z2pi9p9cvmpxl67wj19rc-python3.13-bitsandbytes-0.48.2/lib/python3.13/site-packages:/nix/store/phh0k4akwxnv25dinavhdj3bwqn627q5-python3.13-trove-classifiers-2025.9.11.17/lib/python3.13/site-packages:/nix/store/m8qc4j0vhq56hzjzmadm92g807x2c74d-python3.13-setproctitle-1.3.7/lib/python3.13/site-packages:/nix/store/fc7y1gcck7pw14qsz0n26iwi17wsb5ka-python3.13-openai-harmony-0.0.8/lib/python3.13/site-packages:/nix/store/c0ipi7k8vjdxhibas4320n8gg3l1wz12-python3.13-httptools-0.6.4/lib/python3.13/site-packages:/nix/store/4lj3364zlvlf1gni54q0rx12wm25rn2f-python3.13-python-dotenv-1.1.1/lib/python3.13/site-packages:/nix/store/l72p1ljpqmm419b1im6dj75x505mpd46-python3.13-uvloop-0.22.0/lib/python3.13/site-packages:/nix/store/kv15hmg5xx82frzfh6gbk7kgdlwjqxlc-python3.13-py-libnuma-1.2/lib/python3.13/site-packages:/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/lib/python313.zip:/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/lib/python3.13:/nix/store/7gl4khq26h625mpqzkiiqsify3hclar1-python3-3.13.9/lib/python3.13/lib-dynload:/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/_vendor', '-DFETCHCONTENT_BASE_DIR=/build/source/.deps', '-DCMAKE_JOB_POOL_COMPILE:STRING=compile', '-DCMAKE_JOB_POOLS:STRING=compile=24']' returned non-zero exit status 1.

ERROR Backend subprocess exited when trying to invoke build_wheel
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.9-cachyos, NixOS, 26.05 (Yarara), 26.05.20251127.2fad6ea`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/fqyqwsw82v7jr6ncgrrg7cama5sl577r-source`

### Notify maintainers

@happysalada 
@CertainLach 
@daniel-fahey 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/vllm/0006-drop-rocm-extra-reqs.patch
+++ b/pkgs/development/python-modules/vllm/0006-drop-rocm-extra-reqs.patch
@@ -1,0 +1,20 @@
+diff --git a/requirements/rocm.txt b/requirements/rocm.txt
+index 375f0a019..04a59dc3b 100644
+--- a/requirements/rocm.txt
++++ b/requirements/rocm.txt
+@@ -9,4 +9,2 @@ ray[cgraph]>=2.48.0 # Ray Compiled Graph, required for pipeline parallelism in V1.
+ peft
+-pytest-asyncio
+-tensorizer==2.10.1
+ packaging>=24.2
+@@ -12,6 +10,2 @@ tensorizer==2.10.1
+ packaging>=24.2
+-setuptools>=77.0.3,<80.0.0
+-setuptools-scm>=8
+-runai-model-streamer[s3,gcs]==0.15.3
+-conch-triton-kernels==1.2.1
+ timm>=1.0.17
+ grpcio-tools>=1.76.0
+@@ -17,2 +13,1 @@ conch-triton-kernels==1.2.1
+ timm>=1.0.17
+-grpcio-tools>=1.76.0

--- a/pkgs/development/python-modules/vllm/default.nix
+++ b/pkgs/development/python-modules/vllm/default.nix
@@ -27,12 +27,14 @@
 
   # dependencies
   aioprometheus,
+  amdsmi,
   anthropic,
   bitsandbytes,
   blake3,
   cachetools,
   cbor2,
   compressed-tensors,
+  datasets,
   depyf,
   einops,
   fastapi,
@@ -58,6 +60,7 @@
   outlines,
   pandas,
   partial-json-parser,
+  peft,
   prometheus-fastapi-instrumentator,
   py-cpuinfo,
   pyarrow,
@@ -70,6 +73,7 @@
   sentencepiece,
   setproctitle,
   tiktoken,
+  timm,
   tokenizers,
   torch,
   torchaudio,
@@ -294,7 +298,7 @@ let
     else if cudaSupport then
       gpuArchWarner supportedCudaCapabilities unsupportedCudaCapabilities
     else if rocmSupport then
-      rocmPackages.clr.gpuTargets
+      rocmPackages.clr.localGpuTargets or rocmPackages.clr.gpuTargets
     else
       throw "No GPU targets specified"
   );
@@ -309,6 +313,13 @@ let
     cuda_nvrtc
     # cusparselt # cusparseLt.h
     libcublas
+  ];
+
+  # header path ends up missing rocthrust & its deps
+  rocmExtraIncludeFlags = lib.concatMapStringsSep " " (pkg: "-I${lib.getInclude pkg}/include") [
+    rocmPackages.rocthrust
+    rocmPackages.rocprim
+    rocmPackages.hipcub
   ];
 
   # Some packages are not available on all platforms
@@ -338,6 +349,7 @@ buildPythonPackage.override { stdenv = torch.stdenv; } (finalAttrs: {
     ./0002-setup.py-nix-support-respect-cmakeFlags.patch
     ./0003-propagate-pythonpath.patch
     ./0005-drop-intel-reqs.patch
+    ./0006-drop-rocm-extra-reqs.patch
   ];
 
   postPatch = ''
@@ -413,6 +425,16 @@ buildPythonPackage.override { stdenv = torch.stdenv; } (finalAttrs: {
         rocprim
         hipsparse
         hipblas
+        rocrand
+        hiprand
+        rocblas
+        miopen-hip
+        hipfft
+        hipcub
+        hipsolver
+        rocsolver
+        hipblaslt
+        rocm-runtime
       ]
     )
     ++ lib.optionals stdenv.cc.isClang [
@@ -485,6 +507,13 @@ buildPythonPackage.override { stdenv = torch.stdenv; } (finalAttrs: {
     cupy
     flashinfer
     nvidia-ml-py
+  ]
+  ++ lib.optionals rocmSupport [
+    rocmPackages.rocminfo
+    amdsmi
+    datasets
+    peft
+    timm
   ];
 
   optional-dependencies = {
@@ -523,9 +552,12 @@ buildPythonPackage.override { stdenv = torch.stdenv; } (finalAttrs: {
     }
     // lib.optionalAttrs rocmSupport {
       VLLM_TARGET_DEVICE = "rocm";
-      # Otherwise it tries to enumerate host supported ROCM gfx archs, and that is not possible due to sandboxing.
-      PYTORCH_ROCM_ARCH = lib.strings.concatStringsSep ";" rocmPackages.clr.gpuTargets;
-      ROCM_HOME = "${rocmPackages.clr}";
+      PYTORCH_ROCM_ARCH = gpuTargetString;
+      # vLLM's CMake logic checks `ROCM_PATH` to decide whether HIP/ROCm is available.
+      ROCM_PATH = "${rocmPackages.clr}";
+      TRITON_KERNELS_SRC_DIR = "${lib.getDev triton-kernels}/python/triton_kernels/triton_kernels";
+      HIPFLAGS = rocmExtraIncludeFlags;
+      CXXFLAGS = rocmExtraIncludeFlags;
     }
     // lib.optionalAttrs cpuSupport {
       VLLM_TARGET_DEVICE = "cpu";

--- a/pkgs/development/python-modules/vllm/default.nix
+++ b/pkgs/development/python-modules/vllm/default.nix
@@ -590,6 +590,7 @@ buildPythonPackage.override { stdenv = torch.stdenv; } (finalAttrs: {
       happysalada
       lach
       daniel-fahey
+      LunNova # esp. for ROCm
     ];
     badPlatforms = [
       # CMake Error at cmake/cpu_extension.cmake:188 (message):

